CMakeLists.txt

```diff
@@ -0,0 +1,199 @@
+# We need 3.12 or later, so that we can set policy CMP0074; see below.
+cmake_minimum_required(VERSION 3.12)
+
+# We want find_path() and find_library() to honor {packagename}_ROOT
+if(POLICY CMP0074)
+  cmake_policy(SET CMP0074 NEW)
+endif()
+
+if(POLICY CMP0111)
+  cmake_policy(SET CMP0111 NEW)
+endif()
+
+# MAIN_PROJECT CHECK
+set(MAIN_PROJECT OFF)
+if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+  set(MAIN_PROJECT ON)
+endif()
+
+# Include git module required for version
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules/")
+include(VersionFromGit)
+
+# Get the git version
+version_from_git(LOG ON TIMESTAMP "%Y%m%d%H%M%S")
+
+project(
+  PcapPlusPlus
+  DESCRIPTION
+    "PcapPlusPlus is a multiplatform C++ library for capturing, parsing and crafting of network packets."
+  VERSION ${VERSION})
```

Comment:
Does `${VERSION}` come from the `VersionFromGit` module?
The latest version is maintained in [`PcapPlusPlusVersion.h`](https://github.com/seladb/PcapPlusPlus/blob/master/Common%2B%2B/header/PcapPlusPlusVersion.h), however in order to make things simpler we can just hard-code it here, and maybe change `PcapPlusPlusVersion.h` to take the version from it?

Reply:
Getting in from git allow you to know if the environement is clean or dirty.
And if you are on top of a version you get how many commit and the exact sha1 in the compiled executable.
Moreover the only human actions is now only to "git tag" which avoid human errors.